### DISCUSSION TITLE
How limit price is executed.

### DISCUSSION THREAD

**Discussion by kratos143** [kernc/backtesting.py#894] [Q&A] [answered]
Suppose i have 5 minutes data frame. And i want to enter a short position if previous candle low is broken. I don't want to wait for current candle to close. I want to enter when the low is broken. Can we do this like limit = self.data.Low[-1]

**Comment by kernc** [comment, marked as answer]
When you "see" the current bar, it has already closed. This is how the framework works, and this prevents us from falling for lookahead bias.

The way to achieve this is as you say, placing a stop-limit order. If you just placed a limit order, it would execute whenever the price is above the set limit (for short orders), so you need to add a stop to make sure the order only becomes valid once the stop is hit. I.e.:
```py
low = self.data.Low[-1]
self.sell(limit=low, stop=low, ...)
```

**Reply by kratos143** [reply]
Thanks for the response. I approached that problem in different way. actually I have alert candle in my dataframe. So I want to enter at alert candle low. I just changed the close[-1] to low[-2] . That solved my entry price issue. We can add slippages and all using commissions so it is working for me.
